### ISSUE TITLE
import/no-duplicates false positives for TypeScript module declarations

### ISSUE THREAD

**Issue by remcohaszing** [import-js/eslint-plugin-import#2273]
The `import/no-duplicates` rule gives false positives for imports inside TypeScript module declarations.

Practical example:

```ts
declare module 'monaco-editor/esm/vs/editor/contrib/documentSymbols/documentSymbols' {
  import { ITextModel, languages } from 'monaco-editor';
}

declare module 'monaco-editor/esm/vs/editor/editor.worker' {
  import { worker } from 'monaco-editor';
}
```

This is happening, because `import/no-duplicates` collects all import statements and assumes they are in the program body.

I would expect the `import/no-duplicates` rule to be applied to each body separately. Currently these are the `Program` body and the `TSModuleBlock` body, but who knows what the future holds.

Alternatively the rule could be applied to imports which are direct children `Program` body only.

This is the same as #2217, but for a different rule. This means I can probably fix this if it gets accepted. :)
